### PR TITLE
feat(components): [dialog] append dialog to any element

### PR DESCRIPTION
**Pull request by Karolis-Stoncius**
Add the ability to append dialog not just to body, but to any element

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at 81e0ce3</samp>

This pull request adds a new feature to the Dialog component that allows users to customize the element that the Dialog appends to. It updates the component's documentation, script, and template to support the new `append-to` attribute.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at 81e0ce3</samp>

* Add a new attribute `append-to` to the Dialog component, allowing users to specify which element the Dialog appends to ([link](https://github.com/element-plus/element-plus/pull/14317/files?diff=unified&w=0#diff-abbb3d078da77db6d9ad1eed9a9356891f9119733d8e8bb4fc8771e3429dc626R128), [link](https://github.com/element-plus/element-plus/pull/14317/files?diff=unified&w=0#diff-8e4884ffc45253766cb46796a84d2aeb1fb07f95966b3a4e2440f264948807b6R17-R23), [link](https://github.com/element-plus/element-plus/pull/14317/files?diff=unified&w=0#diff-e9f41878502088bc1f6f1f49029526fcd3fa71df716e6b4dd199947c046b469dL2-R2))
* Modify the Dialog component template to use the `appendTo` prop as the target of the teleport directive, and disable the teleport if both `appendToBody` and `appendTo` are falsy ([link](https://github.com/element-plus/element-plus/pull/14317/files?diff=unified&w=0#diff-e9f41878502088bc1f6f1f49029526fcd3fa71df716e6b4dd199947c046b469dL2-R2))
* Update the `dialog.vue` and `dialog.ts` files to reflect the new attribute and prop ([link](https://github.com/element-plus/element-plus/pull/14317/files?diff=unified&w=0#diff-e9f41878502088bc1f6f1f49029526fcd3fa71df716e6b4dd199947c046b469dL2-R2), [link](https://github.com/element-plus/element-plus/pull/14317/files?diff=unified&w=0#diff-8e4884ffc45253766cb46796a84d2aeb1fb07f95966b3a4e2440f264948807b6R17-R23))

### CODE REVIEW

**Comment by Karolis-Stoncius** [comment]
This is the same PR as [this](https://github.com/element-plus/element-plus/pull/13819), I had to delete my fork, so I created a new PR.

**Comment by Karolis-Stoncius** [comment]
@btea I'm not sure how to fix the failing tests. When I run them locally a bunch of other tests fail as well, even if I revert the changes

**Comment by Karolis-Stoncius** [comment]
@btea any updates on this?

File: packages/components/dialog/src/dialog.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <teleport to="body" :disabled="!appendToBody">
+  <teleport :to="appendTo" :disabled="!appendToBody && !appendTo">
```
Comment (btea) [inline packages/components/dialog/src/dialog.vue:2]: The value of disabled is `true` before modification and then it is `false`. You should use `||`.

Comment (Karolis-Stoncius) [reply packages/components/dialog/src/dialog.vue:2]: Isn't it `undefined` by default? https://github.com/element-plus/element-plus/blob/e32cef1d1b269ea16752b0ac15c5bab41fae9038/packages/components/dialog/src/dialog.ts#L15

Comment (Karolis-Stoncius) [reply packages/components/dialog/src/dialog.vue:2]: Isn't it `false` by default? https://github.com/element-plus/element-plus/blob/e32cef1d1b269ea16752b0ac15c5bab41fae9038/packages/components/dialog/src/dialog.ts#L15

Comment (Karolis-Stoncius) [reply packages/components/dialog/src/dialog.vue]: I found a fix, seems like tests are passing successfully now

File: packages/components/dialog/src/dialog.vue [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <template>
-  <teleport to="body" :disabled="!appendToBody">
+  <teleport
+    :to="appendTo"
+    :disabled="appendTo !== 'body' ? false : !appendToBody"
+  >
```
Comment (btea) [inline packages/components/dialog/src/dialog.vue:5]: The logic here seems a bit weird, maybe just keep an `append-to` prop, but removing `appnedToBody` now would also be a breaking change.
Let's keep it like this for now.

**Review by btea — APPROVED** [review]

**Comment by Karolis-Stoncius** [comment]
@btea Hi, when will this be released?

**Comment by btea** [comment]
Don't worry, it still needs to be reviewed by others.

**Review by ryuhangyeong — APPROVED** [review]

File: docs/en-US/component/dialog.md [hunk 3]
```diff
@@ -125,6 +125,7 @@ When using `modal` = false, please make sure that `append-to-body` was set to **
 | modal                      | whether a mask is displayed                                                                          | ^[boolean]                          | true    |
 | modal-class                | custom class names for mask                                                                          | ^[string]                           | —       |
 | append-to-body             | whether to append Dialog itself to body. A nested Dialog should have this attribute set to `true`    | ^[boolean]                          | false   |
+| append-to                  | which element the Dialog appends to. Will override `append-to-body`                                  | ^[string]                           | body    |
```
Comment (btea) [inline docs/en-US/component/dialog.md:128]: Perhaps a version tag `^(2.4.3)` needs to be added here.

Comment (Karolis-Stoncius) [reply docs/en-US/component/dialog.md]: Added

**Review by btea — APPROVED** [review]

**Pull request merged**